src/world/components/lib/checkTransitionMatch.js

```diff
@@ -0,0 +1,14 @@
+const _ = require('lodash');
+const Type = require('../../../components/type.js');
+
+module.exports = function checkTransitionMatch(rule, thing, locations, action) {
+  const ruleSource = rule.getSource();
+  const targetLocation = rule.getConsequentTarget();
+  const sourceMatch = ruleSource instanceof Type
```

Comment:
rule.getSource() should be whatever is in the "source" position of rule.cause.type. So that could be either a type or the id of a thing if someone is manually creating events.

So in this function we have a rule, a thing, the locations for that thing other than the one it's in, and the action type "move_out".

sourceMatch says: can this rule be caused by this thing?
originMatch says: the rule says move from point A to point B, is "thing" currently located in point A?
destinationMatch says:  is point B contained in the set of possible locations for this thing?
actionMatch: checks to make sure the rule action is "move out", which is the action constant for location transitions.

Perhaps possibleLocations could be calculated inside this function instead of in the parent function "matchTransitionFor"?